src/Data/Validation/Internal.hs

```diff
@@ -0,0 +1,76 @@
+module Data.Validation.Internal where
+
+------------------------------------------------------------------------------------------------------------------------
+import Prelude
+import Data.Map
+import Language.Haskell.TH (Name)
+------------------------------------------------------------------------------------------------------------------------
+
+-- | A type that holds aggregated validation failures.
+data VCtx f a 
+  -- | A value that is assumed to be valid.
+  = ValidCtx a 
+  -- | A value that has failures but can continue to be validated.
+  | DisputedCtx [f] (Map [Name] [f]) a 
+  -- | A that has failures and cannot be validated further.

```

Comment:
*A value that*

Reply:
I'm not sure what you are trying to point out here.

Reply:
"A that has" -> "A value that has"

Reply:
Oh, wow. I see it now...